src/rust-crypto/backup.ts

```diff
@@ -207,15 +207,18 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
             }
             keysByRoom.get(roomId)!.set(key.session_id, key);
         }
-        await this.olmMachine.importBackedUpRoomKeys(keysByRoom, (progress: BigInt, total: BigInt): void => {
-            const importOpt: ImportRoomKeyProgressData = {
-                total: Number(total),
-                successes: Number(progress),
-                stage: "load_keys",
-                failures: 0,
-            };
-            opts?.progressCallback?.(importOpt);
-        });
+        await this.olmMachine.importBackedUpRoomKeys(
+            keysByRoom,
+            (progress: BigInt, total: BigInt, failures?: BigInt): void => {
+                const importOpt: ImportRoomKeyProgressData = {
+                    total: Number(total),
+                    successes: Number(progress),
+                    stage: "load_keys",
+                    failures: failures == undefined ? 0 : Number(failures),
```

Comment:
Do we really need this conditional? Won't `failures` always be defined?

Reply:
`failures` will be `undefined` if it's called from an older version of matrix-sdk-crypto-wasm.  I guess we can drop it after the `package.json` depends on the latest version.

Reply:
I think the easiest thing to do would be to do the upgrade first, *then* land this PR.

Reply:
Fair enough.  I'll mark this as draft again until https://github.com/matrix-org/matrix-js-sdk/pull/4021 is merged, and then make the changes